zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/OutboundCallHandler.kt

```diff
@@ -42,21 +43,24 @@ internal class OutboundCallHandler(
   ): Any? {
     val function = functionsList[functionIndex]
     val argsList = args.toList()
-    val call = RealCall(
+    val internalCall = InternalCall(
       serviceName = serviceName,
-      serviceOrNull = service,
       functionName = function.name,
       function = function,
       args = argsList
     )
-    val encodedCall = endpoint.json.encodeToStringFast(endpoint.callSerializer, call)
-
-    val callStart = endpoint.eventListener.callStart(call)
-    val encodedResult = endpoint.outboundChannel.call(encodedCall)
-
-    val result = endpoint.json.decodeFromStringFast(function.kotlinResultSerializer, encodedResult)
-    endpoint.eventListener.callEnd(call, result, callStart)
-    return result.getOrThrow()
+    val externalCall = endpoint.callCodec.encodeCall(internalCall, service)
+    val callStart = when (service) {
```

Comment:
I wonder if a `fun Service.ifNotSuspendCallback(block: Unit) {}` would work better here than the 4 when blocks I've seen so far

Reply:
Ooooh I’m generally reluctant to introduce control flow things like that? But it does sound tempting